cmd/oras/internal/option/target.go

```diff
@@ -86,6 +87,9 @@ func (opts *Target) Parse() error {
 	switch {
 	case opts.isOCILayout:
 		opts.Type = TargetTypeOCILayout
+		if len(opts.headerFlags) != 0 {
+			return errors.New("customer header([--{from|to}]-header/H) cannot be used on OCI image layout target")
```

Comment:
```suggestion
			return errors.New("customer header flags cannot be used on an OCI image layout target")
```
What you have is more correct, but maybe this would do.